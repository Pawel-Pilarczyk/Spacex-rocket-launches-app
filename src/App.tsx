import React from 'react';
import {ApolloProvider, ApolloClient, InMemoryCache} from '@apollo/client';
import RootNavigation from './navigation';

export const API = 'https://api.spacex.land/graphql';

const client = new ApolloClient({
  uri: API,
  cache: new InMemoryCache(),
});

const App = () => {
  return (
    <ApolloProvider client={client}>
      <RootNavigation />
    </ApolloProvider>
  );
};

export default App;
