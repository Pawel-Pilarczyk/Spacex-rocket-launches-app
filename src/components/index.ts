export {default as Button} from './Button';
export {default as Typography} from './Typography';
export {default as Input} from './Input';
export {default as Item} from './Item';
export {default as ModalLoader} from './Modals/ModalLoader';
