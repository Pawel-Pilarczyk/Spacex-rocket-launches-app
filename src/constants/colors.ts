export const COLORS = {
  WHITE: '#fff',
  BLACK: '#000',
  PRIMARY: '#1C6DD0',
  PRIMARYDARK: '#08458F',
  SUCCESS: '#22C55E',
  ERROR: '#DC2626',
  DISABLED: '#8f8f8f',
};
