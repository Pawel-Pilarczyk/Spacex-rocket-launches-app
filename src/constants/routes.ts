export const ROUTES = {
  WELCOME: 'WELCOME',
  HOME: 'HOME',
  DETAILS: 'DETAILS',
  ERROR: 'ERROR',
};
