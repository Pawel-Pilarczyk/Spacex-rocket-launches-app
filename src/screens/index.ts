export {default as Home} from './Home';
export {default as Welcome} from './Welcome';
export {default as LaunchDetails} from './LaunchDetails';
export {default as ErrorScreen} from './Error';
