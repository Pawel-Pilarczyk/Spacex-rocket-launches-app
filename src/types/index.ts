export type TNavigation = {
  navigate: (route: string, params?: any) => void;
  goBack: () => void;
};

export type TRoute = {
  params: any;
};

export type TScreenProps = {
  navigation: TNavigation;
  route?: TRoute;
};

export type TLaunch = {
  mission_name: string;
  launch_site: {
    site_name_long: string;
  };
  links: {
    article_link: string;
    video_link: string;
  };
  rocket: {
    rocket_name: string;
  };
  launch_date_utc: string;
  launch_success: boolean;
  id: string;
};

export type TLaunchesData = {
  launchesPast: Array<TLaunch>;
};
