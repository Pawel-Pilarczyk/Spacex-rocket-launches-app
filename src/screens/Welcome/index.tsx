import {StyleSheet, View} from 'react-native';
import React from 'react';
import {COLORS} from 'src/constants/colors';
import {ROUTES} from 'src/constants/routes';
import {scaling} from 'src/styles/scaling';
import {Rocket} from 'src/assets/svg';
import {Typography, Button} from 'src/components';
import {TScreenProps} from 'src/types';

const Welcome = ({navigation}: TScreenProps) => {
  const handleNavigation = () => navigation.navigate(ROUTES.HOME);
  return (
    <View style={styles.wrapper}>
      <Typography size="40" type="bold">
        Welcome
      </Typography>
      <Typography size="18" style={styles.paragraph}>
        Follow the latest SpaceX rocket launches with just one click
      </Typography>
      <Rocket width={'75%'} height={'60%'} testID="rocketSVG" />
      <Button style={styles.button} onPress={handleNavigation}>
        Let's get Started!
      </Button>
    </View>
  );
};

export default Welcome;

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    backgroundColor: COLORS.WHITE,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: scaling.hs(10),
    paddingVertical: scaling.vs(25),
  },
  button: {
    width: '100%',
    marginTop: 'auto',
  },
  paragraph: {
    marginTop: scaling.vs(20),
    marginBottom: scaling.vs(40),
    textAlign: 'center',
  },
});
