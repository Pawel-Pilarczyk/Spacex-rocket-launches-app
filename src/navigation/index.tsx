import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {ROUTES} from 'src/constants/routes';
import {Home, Welcome, LaunchDetails, ErrorScreen} from 'src/screens';

const RootStack = createStackNavigator();

const RootNavigation = () => {
  return (
    <NavigationContainer>
      <RootStack.Navigator
        screenOptions={{
          headerTitleAlign: 'center',
          headerBackTitle: '',
        }}>
        <RootStack.Screen
          name={ROUTES.WELCOME}
          component={Welcome}
          options={{headerShown: false}}
        />
        <RootStack.Screen
          name={ROUTES.HOME}
          component={Home}
          options={{headerTitle: 'Missions List'}}
        />
        <RootStack.Screen
          name={ROUTES.DETAILS}
          component={LaunchDetails}
          options={{headerTitle: 'Mission Details'}}
        />
        <RootStack.Screen name={ROUTES.ERROR} component={ErrorScreen} />
      </RootStack.Navigator>
    </NavigationContainer>
  );
};

export default RootNavigation;
