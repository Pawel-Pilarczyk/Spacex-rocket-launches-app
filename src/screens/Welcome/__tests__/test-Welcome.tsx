/* eslint-disable @typescript-eslint/no-unused-vars */
/**
 * @format
 */

import 'react-native';
import React from 'react';
import Welcome from 'src/screens/Welcome';
import {TNavigation} from 'src/types';
import {ROUTES} from 'src/constants/routes';

// Note: test renderer must be required after react-native.
import {render, fireEvent} from '@testing-library/react-native';
const navigation: TNavigation = {
  navigate: (route: string, params?: any) => {},
  goBack: () => {},
};
it('shoukd render correctly', () => {
  const welcome = render(<Welcome navigation={navigation} />).toJSON();
  expect(welcome).toMatchSnapshot();
});

it('should go to main Home page after pressing button Lets get started', () => {
  jest.spyOn(navigation, 'navigate');
  const welcomePage = render(<Welcome navigation={navigation} />);
  const continueButton = welcomePage.getByText("Let's get Started!");
  fireEvent.press(continueButton);
  expect(navigation.navigate).toHaveBeenCalledWith(ROUTES.HOME);
});

it('Should have a child of SVG graphic', () => {
  const welcome = render(<Welcome navigation={navigation} />);
  const svg = welcome.getByTestId('rocketSVG');
  expect(svg).toBeDefined();
});
