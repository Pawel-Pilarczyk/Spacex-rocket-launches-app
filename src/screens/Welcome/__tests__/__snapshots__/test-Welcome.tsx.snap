// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoukd render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
      "paddingVertical": 41,
    }
  }
>
  <Text
    style={
      Array [
        undefined,
        Object {
          "color": "#000",
          "fontSize": 80,
          "fontWeight": "800",
        },
      ]
    }
  >
    Welcome
  </Text>
  <Text
    style={
      Array [
        Object {
          "marginBottom": 66,
          "marginTop": 33,
          "textAlign": "center",
        },
        Object {
          "color": "#000",
          "fontSize": 36,
          "fontWeight": "400",
        },
      ]
    }
  >
    Follow the latest SpaceX rocket launches with just one click
  </Text>
  <rocket-svgrepo-com.svg
    height="60%"
    testID="rocketSVG"
    width="75%"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1C6DD0",
        "borderRadius": 26,
        "flex": 1,
        "height": 92,
        "justifyContent": "center",
        "marginTop": "auto",
        "maxHeight": 92,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          undefined,
          Object {
            "color": "#fff",
            "fontSize": 36,
            "fontWeight": "400",
          },
        ]
      }
    >
      Let's get Started!
    </Text>
  </View>
</View>
`;
